src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double lateralSide;
+    private double upperBase;
+    private double bottomBase;
+    private double height;
+    private double triangleLeg;
+
+    public IsoscelesTrapezoid() {
+    }
+
+    public IsoscelesTrapezoid(double upperBase, double bottomBase, double lateralSide) {
+        this.typeOfFigure = TypeOfFigure.ISOSCELES_TRAPEZOID;
+        this.area = calculateArea(upperBase, bottomBase, triangleLeg, height);
```

Comment:
Creating a figure, don't pass expressions in the constructor. Instead, create separate variables and pass them on for better code readability. Moreover, the `triangleLeg` and `height` parameters are not initialized before they are used in `calculateArea` method call.